components/brave_account/brave_account_service.cc

```diff
@@ -253,4 +263,90 @@ std::optional<mojom::RegisterErrorCode> BraveAccountService::TransformError(
                                              : std::nullopt;
 }
 
+void BraveAccountService::OnVerificationTokenChanged() {
+  if (pref_service_->GetString(prefs::kVerificationToken).empty()) {
+    return CHECK_DEREF(verify_result_timer_.get()).Stop();
+  }
+
+  ScheduleVerifyResult();
+}
+
+void BraveAccountService::ScheduleVerifyResult(base::TimeDelta delay) {
+  CHECK_DEREF(verify_result_timer_.get())
+      .Start(FROM_HERE, delay,
```

Comment:
please stop the timer on `::Shutdown()`

Reply:
https://github.com/brave/brave-core/pull/31896#discussion_r2461144046